text/0039-_formatted-field-behavior.md

```diff
@@ -0,0 +1,473 @@
+
+**Example 6**
+
+Given these search parameters:
+```
+{
+    "q": "prince",
+    "attributesToRetrieve": ["title"],
+    "attributesToHighlight": ["*"]
+}
+```
+
+As a user i get:
+```
+{
+    "hits": [
+        {
+            "title": "Prince Avalanche",
+            "_formatted": {
+                "title": "<em>Prince</em> Avalanche"
+            }
+        }
+    ]
+}
+```
+
+`_formatted` is only filled with the `attributesToRetrieve` fields despite the fact that the user may wants all fields to be in `_formatted` and be highlighted or cropped given `attributesToHighlight` or `attributesToCrop` values.
+
+#### Current Transplant/Milli Behavior (0.21)
+
+**Example 1**
+
+Given these search parameters:
+```
+{
+    "q": "prince",
+    "attributesToRetrieve": ["*"]
+}
+```
+
+As a user i get:
+
+```
+{
+    "hits": [
+        {
+            "title": "Prince Avalanche",
+            "actor": "Prince",
+            "poster": "https://image.tmdb.org/t/p/w1280/3KHiQt54usbHyIjLIMzaDAoIJNK.jpg"
+        }
+    ]
+}
+```
+
+Since `attributesToHighlight` and `attributesToCrop` are not set, `_formatted` is not computed.
+
+Same behavior as v0.20 release.
+
+**Example 2**
+
+Given these search parameters:
+```
+{
+    "q": "prince",
+    "attributesToRetrieve": ["*"],
+    "attributesToHighlight": ["wrongFieldName"]
+}
+```
+
+As a user i get:
+
+```
+{
+    "hits": [
+        {
+            "title": "Prince Avalanche",
+            "poster": "https://image.tmdb.org/t/p/w1280/3KHiQt54usbHyIjLIMzaDAoIJNK.jpg",
+        }
+    ]
+}
+```
+
+Unlike v0.20, if no field are matched to be formatted, `_formatted` is not computed nor returned in response.
+
+**Example 3**
+
+Given these search parameters:
+
+```
+{
+    "q": "Prince",
+    "attributesToRetrieve": ["title"],
+    "attributesToHighlight": ["actor"]
+}
+```
+
+As a user i get:
+
+```
+{
+    "hits": [
+        {
+            "title": "Prince Avalanche",
+            "_formatted": {
+                "actor": "<em>Prince</em>"
+            }
+        }
+    ]
+}
+```
+
+`_formatted` is only filled with the fields set in `attributesToHighlight` despite the fact that the user only ask for `title` in `attributesToRetrieve`.
+
+Same behavior as v0.20 release.
+
+**Example 4**
+
+Given these search paramters:
+
+```
+{
+    "q": "Prince",
+    "attributesToRetrieve": ["actor", "title"],
+    "attributesToHighlight": ["actor"]
+}
+```
+
+As a user i get:
+
+```
+{
+    "hits": [
+        {
+            "title": "Prince Avalanche",
+            "actor": "Prince",
+            "_formatted": {
+                "actor": "<em>Prince</em>"
+            }
+        }
+    ]
+}
+```
+
+Same behavior as v0.20 release.
+
+**Example 5**
```

Comment:
Can you precise in this example that put `attributesToRetrieve` as parameter is equivalent to not set `attributesToRetrieve` at all since the default value of `attributesToRetrieve` is `["*"]`?
It might be obvious but it is an important thing we should not forget

Reply:
I will add that part in the Expected MeiliSearch Behavior part! Thanks